### PR TITLE
Results race page 2

### PR DESCRIPTION
**Pull request by glossopm**

### CODE REVIEW

File: routes/results/results.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from flask import render_template, Blueprint, url_for, redirect, request, flash
```
Comment (MikeSilverstone) [inline routes/results/results.py]: request is an unused import

File: routes/results/results.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        if not result_row:
+            db.session.add(RaceResult(position=race_results_form.position.data, time_to_finish=race_results_form.time_to_finish.data, did_not_finish=race_results_form.did_not_finish.data, id_race_participants=race_results_form.id_race_participants.data ))
+            db.session.commit()
+            flash("Race Result recorded for Race Participant ID {}.".format(race_results_form.id_race_participants.data))
+        else:
+            flash("Race Result Failed. Race Result already exists for for Race Participant ID {}.".format(race_results_form.id_race_participants.data))
+        return redirect(url_for('result.race', round_id=round_id, race_id=race_id))  
+    elif race_results_form.errors.items():
+        flash("Form submission not valid. Please resubmit.")
+        return redirect(url_for('result.race', round_id=round_id, race_id=race_id))
```
Comment (MikeSilverstone) [inline routes/results/results.py]: Convention in this codebase is to have newlines after returns and before/after if statements.

File: routes/results/results.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    return render_template('results.html', race=race, participants=participants, race_results_form=race_results_form)
+
+
+def validation(form):
+    return form.validate()
```
Comment (MikeSilverstone) [inline routes/results/results.py]: needs a newline at the end of file.

File: tests/test_results.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        self.client = app.test_client()
+
+    @mock.patch('flask_login.utils._get_user')
+    def test_results_page_response_is_200(self, current_user):
+        with self.client as client:
+            current_user.is_authenticated = True
+            response = client.get("/rounds/1/races/1")
+            self.assertEqual(200, response.status_code)
+
+
```
Comment (MikeSilverstone) [inline tests/test_results.py]: we don't need two newlines here

File: tests/test_results.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    
+    @mock.patch('flask_login.utils._get_user')
+    def test_results_page_displayed(self, current_user):
+        with self.client as client:
+            current_user.is_authenticated = True
+            response = client.get("/rounds/1/races/1")
+            self.assertIn(b"Time to Finish", response.data)
+
+    @mock.patch("db.models.RaceParticipants.get_race_participants_race_id", MagicMock(return_value=[MockRaceParticipants(1,1,1), MockRaceParticipants(2,2,1)]))
```
Comment (MikeSilverstone) [inline tests/test_results.py]: spaces after commas

Comment (MikeSilverstone) [reply tests/test_results.py]: and this is over 80 characters long!

**Review by MikeSilverstone — DISMISSED** [review]
Looks good, just install a linter. It'll save both of us a lot of time.

**Review by infinitydeep — APPROVED** [review]
Functionality on point, tests all valuable and passing. No bugs found

File: tests/test_results.py [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+                MagicMock(return_value=[MockRaceParticipants(1, 1, 1),
+                                        MockRaceParticipants(3, 3, 1)]))
+    @mock.patch("db.models.Race.get_race", 
+                MagicMock(return_value=MockRace(1,
+                                                datetime.datetime.now(), 
+                                                'TEST_STATUS', 
+                                                1)))
+    @mock.patch('flask_login.utils._get_user')
+    @mock.patch("forms.forms.RaceResultsForm")
+    def test_results_valid_data_posted(self, current_user, race_results_form):
```
[the remark this answers is not in the document]
Comment (sorsby) [reply tests/test_results.py]: @MikeSilverstone fix it

**Pull request merged**